### PR TITLE
[MI-123] Add group memberships resource and test

### PR DESCRIPTION
**Pull request by joseconsador**
[MI-123] 
Add group memberships resource and test

/cc @miogalang @jwswj @mmolina @atroche 

### References
 - Jira link: https://zendesk.atlassian.net/browse/MI-123

### Risks
 - Low. We might not be able to call the group memberships endpoints

### CODE REVIEW

File: src/Zendesk/API/Resources/Groups.php [hunk 1]
```diff
@@ -20,6 +20,16 @@ class Groups extends ResourceAbstract
     /**
      * {@inheritdoc}
      */
+    public static function getValidSubResources()
+    {
+        return [
+            'groupMemberships' => GroupMemberships::class,
```
Comment (miogalang) [inline src/Zendesk/API/Resources/Groups.php]: maybe just memberships? `$client->groups()->memberships()` sounds right

Comment (joseconsador) [inline src/Zendesk/API/Resources/Groups.php]: :haircut:

**Comment by miogalang** [comment]
:+1:

**Comment by jwswj** [comment]
:+1:

**Pull request merged**